src/extent.js

```diff
@@ -1,24 +1,16 @@
+import accessor from './accessor';
+
 export default function(array, f) {
   var i = -1,
       n = array.length,
       a,
       b,
       c;
 
-  if (f == null) {
-    while (++i < n) if ((b = array[i]) != null && b >= b) { a = c = b; break; }
-    while (++i < n) if ((b = array[i]) != null) {
-      if (a > b) a = b;
-      if (c < b) c = b;
-    }
-  }
-
-  else {
-    while (++i < n) if ((b = f(array[i], i, array)) != null && b >= b) { a = c = b; break; }
-    while (++i < n) if ((b = f(array[i], i, array)) != null) {
-      if (a > b) a = b;
-      if (c < b) c = b;
-    }
+  while (++i < n) if ((b = accessor(f, i, array)) != null && b >= b) { a = c = b; break; }
```

Comment:
thanks your advice. In the process of coding, I tried to use the `console.time()` and `console.timeEnd()` to watch the performance, and result was little difference between after and before. So..I guess whether the data amount in the test is so little?